### PR TITLE
https://github.com/OpenHFT/Chronicle-Queue/issues/822

### PR DESCRIPTION
**Pull request by JerryShea**
Index entry required if first item in a queue is metadata

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/queue/ChronicleQueueIndexTest.java [hunk 1]
```diff
@@ -292,4 +293,157 @@ public void writeReadMetadata() {
             }
         }
     }
+
+    // https://github.com/OpenHFT/Chronicle-Queue/issues/822
+    private void driver0(String[] strings, boolean[] meta, long millis) {
+
+        assert (strings.length == meta.length);
+
+        File tmp = new File("/dev/shm/Issue882");
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/ChronicleQueueIndexTest.java]: not cross platform - I would use getTmpDir

Comment (JerryShea) [reply src/test/java/net/openhft/chronicle/queue/ChronicleQueueIndexTest.java]: other than this minor change this all LGTM although I would not rely on just me ;)

File: src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java [hunk 2]
```diff
@@ -851,7 +851,7 @@ void cleanupStoreFilesWithNoData() {
                     // file not found.
                     if (store == null)
                         break;
-                    if (store.writePosition() == 0 && !store.file().delete() && store.file().exists()) {
+                    if (store.writePosition() == 0 && store.file().exists()) {
                         // couldn't delete? Let's try writing EOF
```
Comment (JerryShea) [inline src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java]: we should change the comment too ;)

File: src/test/java/net/openhft/chronicle/queue/ChronicleQueueIndexTest.java (hunk 1 above)
Comment (rogersimmons) [reply src/test/java/net/openhft/chronicle/queue/ChronicleQueueIndexTest.java]: Thanks @JerryShea  - yes, only intended while testing locally. Swapped to getTmpDir()

File: src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java (hunk 2 above)
Comment (rogersimmons) [reply src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java]: updated

**Review by chronicle-dev — APPROVED** [review]

**Pull request merged**